### PR TITLE
Modify rule S1542: LaYC format

### PR DESCRIPTION
**Pull request by maksim-grebeniuk-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S1542/python/rule.adoc [hunk 1]
```diff
@@ -1,25 +1,39 @@
+This rule raises an issue when function name doesn't respect the naming convention.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S1542/python/rule.adoc:1]: when function name => when a function name

File: rules/S1542/python/rule.adoc [hunk 2]
```diff
@@ -1,25 +1,39 @@
...
+
 == Why is this an issue?
 
 include::../description.adoc[]
 
-=== Noncompliant code example
+According to PEP8 function names should be in all lowercase letters, multiple words should be separated by underscores. This convention is known as "snake_case." For example: ``++calculate_area++``, ``++print_hello++``, ``++process_data++``
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S1542/python/rule.adoc:7]: Slight rewording suggestion:
According to PEP8, function names should be lowercase, with words separated by underscores as necessary to improve readability. This convention...

File: rules/S1542/python/rule.adoc [hunk 3]
```diff
@@ -1,25 +1,39 @@
...
+
+=== How to fix it
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S1542/python/rule.adoc:9]: How to fix it should be a 2nd level title `==`. That said, I think we shouldn't have it at all here, as the rule description is so small (let's keep everything in one single tab).

File: rules/S1542/python/rule.adoc [hunk 4]
```diff
@@ -1,25 +1,39 @@
...
+Respect the naming convention for the function (by default it is the snake case as recommended by PEP8).
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S1542/python/rule.adoc:10]: If we remove the header, then we can rephrase this:
To fix this, respect the naming convention for the function. The default naming convention is snake case, as recommended by PEP8. Other naming conventions can be defined through the `format` rule parameter.

**Review by guillaume-dequenne-sonarsource — CHANGES_REQUESTED** [review]
Some minor suggestions.

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
LGTM.

File: rules/S1542/python/rule.adoc [hunk 5]
```diff
@@ -1,25 +1,38 @@
+This rule raises an issue when a function name doesn't respect the naming convention.
+
 == Why is this an issue?
 
 include::../description.adoc[]
```
Comment (jean-jimbo-sonarsource) [inline rules/S1542/python/rule.adoc:5]: what's going on here?

**Review by jean-jimbo-sonarsource — CHANGES_REQUESTED** [review]
please see my note

File: rules/S1542/python/rule.adoc (hunk 5 above)
Comment (guillaume-dequenne-sonarsource) [reply rules/S1542/python/rule.adoc:5]: This is supposed to include a common description for multiple languages. Is it not working?
To test it, you can clone the project and open a preview of the page in VSCode or IntelliJ, it should resolve these include statements.
Alternatively, to test it in the SQ UI, you can run `rule-api` to generate the actual html in the `sonar-python` project, then build the analyzer, include the jar into a local SonarQube and run it. It's a bit more involved to do it the first time, but makes reviewing a bit easier afterwards, as you can really make sure everything displays properly.

Comment (maksim-grebeniuk-sonarsource) [reply rules/S1542/python/rule.adoc:5]: You can also use the IntelliJ [AsciiDoc](https://plugins.jetbrains.com/plugin/7391-asciidoc) plugin as well.

Comment (jean-jimbo-sonarsource) [reply rules/S1542/python/rule.adoc:5]: Ahh thanks, I got it working now

**Pull request merged**